src/engine/strat_engine/thinpool/thinpool.rs

```diff
@@ -125,8 +125,14 @@ fn coalesce_segs(
     segments
 }
 
+/// Use the lowater event for two purposes: extending the data dev and also
+/// getting an event when a FreeSpaceState threshold is crossed.
+/// Most of the time, this results in just DATA_LOWATER. The two times it
```

Comment:
I'm having difficulty making sense of this explanation. But I can see that, when the available amount decreases to none, the lowater will be 0.1 * data dev size, if still in Good state.